src/ldp/http/metadata/MappedMetadataWriter.ts

```diff
@@ -8,18 +8,18 @@ import { MetadataWriter } from './MetadataWriter';
  * The header value(s) will be the same as the corresponding object value(s).
  */
 export class MappedMetadataWriter extends MetadataWriter {
-  private readonly headerMap: Record<string, string>;
+  private readonly headerMap: [string, string][];
```

Comment:
What is the reason to start storing key/values like this?

Reply:
I realized here https://github.com/solid/community-server/pull/549/files#diff-02203c0238b2c7906d0fac9cf8a5f3389512070339bb35f0423fbec49eaf3d03R13-R18 it would be a waste to repeatedly call `Object.entries`. Also, the decomposition in the constructor ensures the object cannot be changed after construction (minor of course).

Reply:
I personally don't think the the (very minor) performance increase is worth the (very minor) cost of parameter readability, but both of them are so minor it doesn't really matter. 😄